crates/rpc/rpc/src/validation.rs

```diff
@@ -78,6 +80,10 @@ pub struct ValidationApiInner<Provider: ChainSpecProvider, E> {
     executor_provider: E,
     /// Set of disallowed addresses
     disallow: HashSet<Address>,
+    /// Cached state reads to avoid redundant disk I/O across multiple validation attempts.
+    /// Maps block hash to cached reads for that block's state.
+    /// Uses async `RwLock` to safely handle concurrent validation requests.
+    cached_states: Arc<RwLock<HashMap<B256, CachedReads>>>,
```

Comment:
we only really need one instance here, then we don't need to do manual cleanup.
validations only target the head state so we only need to store `(B256, CachedReads)`

Reply:
Got it, I 've removed the `HashMap`